modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/portlet/action/UpdateUserRolesMVCActionCommand.java

```diff
@@ -172,6 +186,73 @@ else if (exception instanceof MembershipPolicyException) {
 		}
 	}
 
+	private void _deleteUserGroupRoleSegmentData(
```

Comment:
Hi @laszlopap,

Adding this method to the ActionCommand does not guarantee it will be called each time a user group role is deleted. If it is deleted via REST API or separate service call this won't be run. 

I recommend you add this to a model listener inside the segments-service module if possible. This way you get 2 things:
1. Ensure it will always be run.
2. Avoid adding a dependency to segments-api in the users-admin-web module.

What do you think?

Reply:
Thanks for the suggestion @4lejandrito !

I have started to move the logic. However I got stuck, and asked you a question on slack.

Regards,
Laci